zcash_client_backend/src/data_api/scanning.rs

```diff
@@ -6,7 +6,9 @@ use zcash_primitives::consensus::BlockHeight;
 /// Scanning range priority levels.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 pub enum ScanPriority {
-    /// Block ranges that have already been scanned have lowest priority.
+    /// Block ranges that are ignored have lowest priority.
+    Ignored,
```

Comment:
There are a bunch of places where we use the Sapling activation height as the first block height that should potentially be scanned. If I understand correctly, this PR is marking the priority of the blocks between the Sapling activation height and the wallet's birthday as ignored, with the intent that they stay permanently ignored.

Wouldn't it be simpler to explicitly store the birthday and use that in place of the Sapling activation height? That would be a simpler data model with fewer states.

(I know that it is possible to have a rollback just after the wallet is initialised that effectively rolls back its birthday. If that ends up being too complex to handle in the approach I'm suggesting then I can see the rationale for doing it this way.)

Reply:
I see the #901 is about doing precisely that. But the issue is that if you do this change first and #901 later, then you've already complicated the data model by adding the "ignored" priority, and can't undo that.

Reply:
I think that the notion of having an ignored range could actually be generally useful; one hypothetical use case is that someone who knows that their wallet has been offline for a long period of time (say, for example, a paper/cold wallet for which they never gave out any addresses) could explicitly inform the wallet that it should ignore the range since the last height that they interacted with it, and thereby avoid scanning costs. So even after #901 I think it's good to keep the `Ignored` case; I just hadn't really thought about it before or I might have included the behavior I'm implementing here in the first draft.